service/src/utils/onchain-data.ts

```diff
@@ -60,21 +65,26 @@ export async function getOnchainCursedSarcophagi(
 ): Promise<SarcophagusData[]> {
   const archSarco: SarcophagusData[] = [];
 
+  inMemoryStore.gracePeriod = inMemoryStore.gracePeriod || await getGracePeriod(web3Interface);
+
   const sarcoIds = await web3Interface.viewStateFacet.getArchaeologistSarcophagi(
     web3Interface.ethWallet.address
   );
+
   sarcoIds.map(async sarcoId => {
     const sarco = await web3Interface.viewStateFacet.getSarcophagus(sarcoId);
     const archStorage = await web3Interface.viewStateFacet.getSarcophagusArchaeologist(
       sarcoId,
       web3Interface.ethWallet.address
     );
 
-    if (sarco.state === SarcophagusState.Exists && !archStorage.unencryptedShard) {
+    // If sarcophagus exists and the unencrypted shard is empty, then it (potentially)
+    // still needs to be unwrapped
+    if ((sarco.state === SarcophagusState.Exists) && archStorage.unencryptedShard === '0x') {
```

Comment:
Solidity defaults! Hah.

JS' "falsy" logic is such a recipe for disaster.

As an aside: Are those extra parentheses needed around `sarco.state === SarcophagusState.Exists`

Reply:
@DarksightKellar thanks those parens arent needed --- removed them